app/routes/devices.$id.controllers.tsx

```diff
@@ -0,0 +1,66 @@
+import {
+  Link,
+  Outlet,
+  useParams,
+  useRouteLoaderData,
+  useSearchParams,
+} from "@remix-run/react";
+import { HardwareTable } from "~/components/HardwareTable";
+import { loader } from "./devices.$id";
+import { EvolverConfigWithoutDefaults } from "client";
+import { CogIcon } from "@heroicons/react/24/outline";
+
+export const handle = {
+  breadcrumb: ({
+    params,
+  }: {
+    params: { id: string; hardware_name: string };
```

Comment:
maybe `hardware_name` is not right here?

Reply:
Yeah will remove that param isn't in the path and isn't used in the breadcrumb.